src/io.c

```diff
@@ -1432,9 +1432,26 @@ _dispatch_fd_entry_create_with_fd(dispatch_fd_t fd, uintptr_t hash)
 #if defined(_WIN32)
 		DWORD dwType = GetFileType((HANDLE)fd);
 		if (dwType == FILE_TYPE_PIPE) {
-			unsigned long value = 1;
-			int result = ioctlsocket((SOCKET)fd, (long)FIONBIO, &value);
-			(void)dispatch_assume_zero(result);
+			if (_dispatch_handle_is_socket((HANDLE)fd)) {
+				unsigned long value = 1;
+				int result = ioctlsocket((SOCKET)fd, (long)FIONBIO, &value);
+				(void)dispatch_assume_zero(result);
+			}
+			else {
```

Comment:
Please coddle the braces.

```suggestion
			} else {
```

Reply:
Fixed in latest